Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp

```diff
@@ -32,6 +32,22 @@
 #include <wtf/Lock.h>
 #include <wtf/PrintStream.h>
 
+#ifdef USE_OPENCDM_BCM_NEXUS_SVP
+
+#include <b_secbuf.h>
+#include <gst_brcm_svp_meta.h>
+
+struct Rpc_Secbuf_Info {
```

Comment:
Agree that it should be abstracted. However I think it should be absrtracted in the gstreamer elements. As the gstreamer element "Decryption" goes hand-in-hand with the gstreamer sink, where there is no more OCDM, but only GStreamer sink from Broadcom.
Why is the Decryptor sink element hard coded in the WebKit sources ? Should it not just be an GSTElement loaded in the gstreamer directory (independent from WebKit?) and being loaded by the playbin if required ? 
But bottemline, abstraction would be very usefull here.

Reply:
If I am not mistaken, OpenCDM has to do different things depending on if the memory is protected or not, am I wrong? That's why I think it should be abstracted there. Regarding GStreamer already handling protected memory, it is the nexus sink plugin, not the general decryptor so I think we are talking about different things here.

The decryptors are not pure GStreamer elements because EME flow is web platform centric (JS, events, etc) and our implementation relies on WebKit objects. These decryptors can only live inside WebKit. Regarding how they are loaded, it does not matter were they are declared, as long as we properly register them (and we do) they are like any other GStreamer element even if they are not with the rest of GStreamer plugins.